source/agora/cli/multi/main.d

```diff
@@ -0,0 +1,305 @@
+
+*******************************************************************************/
+
+private string replaceAddress(string source, string host)
+{
+    auto x1 = source.indexOf("://");
+    auto x2 = source.lastIndexOf(":");
+
+    if ((x2 == -1) || (x2 == x1))
+        x2 = source.length-1;
+
+    if (x1 == -1)
+        x1 = 0;
+    else
+        x1 += 3;
+
+    return source.replace(x1, x2, host);
+}
+
+unittest
+{
+    assert(replaceAddress("http://node-1:2826", "127.0.0.1") == "http://127.0.0.1:2826");
+    assert(replaceAddress("http://node-1", "127.0.0.1") == "http://127.0.0.1");
+}
+
+
+/*******************************************************************************
+
+    Read config from file
+
+    Params:
+      filename = A file name of configurationn of the node
+
+    Returns:
+      The configuration objects that are used through the node
+
+*******************************************************************************/
+
+private Config readFromFile (string filename)
+{
+    CommandLine cmdln;
+    cmdln.config_path = filename;
+    return parseConfigFile(cmdln);
+}
+
+
+/*******************************************************************************
+
+    Get connection address form Config
+
+    Params:
+      config = A configuration objects that are used through the node
+
+    Returns:
+      An address to connect to the node
+
+*******************************************************************************/
+
+private string getNodeAddress (Config config)
+{
+    return format("http://%s:%d", config.node.address, config.node.port);
+}
+
+
+/*******************************************************************************
+
+    Test the connection to the node.
+
+    Params:
+      address = An address of the node
+
+*******************************************************************************/
+
+private void testConection (string address)
+{
+    import core.time;
+    NetworkInfo info;
+    bool success = false;
+    auto timeout = 5000.msecs;
+    auto limit = MonoTime.currTime + timeout;
+
+    while (!success)
+    {
+        auto left = limit - MonoTime.currTime;
+        if (left.isNegative)
+            break;
+
+        try
+        {
+            success = true;
+            auto clients = new RestInterfaceClient!API(address);
+            info = clients.getNetworkInfo();
+            writefln("Connection to %s succeeded.", address);
+        }
+        catch (Exception ex)
+        {
+            success = false;
+            sleep(10.msecs);
+        }
+    }
+
+    if (!success)
+        writefln("Connection to %s failed.", address);
+}
+
+
+/*******************************************************************************
+
+    Create and run a node. And handle HTTP requests.
+
+    This class is required for terminating a node at the end of the main thread.
+
+*******************************************************************************/
+
+public class NodeLauncher
```

Comment:
We can remove this

Reply:
@Geod24 I removed NodeLauncher, 
but only one node is running and the rest is not working.

Reply:
@Geod24 I found a way. I'll organize it and then commit it again.